app/controllers/api/users.php

```diff
@@ -663,6 +665,45 @@ function createUser(string $hash, mixed $hashOptions, string $userId, ?string $e
         $response->dynamic($user, Response::MODEL_USER);
     });
 
+App::put('/v1/users/:userId/labels')
+    ->desc('Update User Labels')
+    ->groups(['api', 'users'])
+    ->label('event', 'users.[userId].update.labels')
+    ->label('scope', 'users.write')
+    ->label('audits.event', 'user.update')
+    ->label('audits.resource', 'user/{response.$id}')
+    ->label('audits.userId', '{response.$id}')
+    ->label('usage.metric', 'users.{scope}.requests.update')
+    ->label('sdk.auth', [APP_AUTH_TYPE_KEY])
+    ->label('sdk.namespace', 'users')
+    ->label('sdk.method', 'updateLabels')
+    ->label('sdk.description', '/docs/references/users/update-user-labels.md')
+    ->label('sdk.response.code', Response::STATUS_CODE_OK)
+    ->label('sdk.response.type', Response::CONTENT_TYPE_JSON)
+    ->label('sdk.response.model', Response::MODEL_USER)
+    ->param('userId', '', new UID(), 'User ID.')
+    ->param('labels', [], new ArrayList(new Text(128, allowList: [...Text::NUMBERS, ...Text::ALPHABET_UPPER, ...Text::ALPHABET_LOWER]), APP_LIMIT_ARRAY_PARAMS_SIZE), 'Array of user labels. Replaces the previous labels. Maximum of 100 labels are allowed, each 128 characters long. Valid chars are a-z, A-Z, and 0-9.')
```

Comment:
@TorstenDittmann @stnguyen90 how will too many labels affect permissions performance? Should we consider a smaller limit?